Confidence/src/main/java/com/spotify/confidence/Producer.kt

```diff
@@ -8,9 +8,14 @@ data class Event(
     val shouldFlush: Boolean = false
 )
 
-interface EventProducer {
+sealed interface Producer {
```

Comment:
If a Producer can produce both events and context changes, I think it's a requirement that the implementor can control the exact order of these. Not sure if that's possible with the proposed design. Maybe an alternative would be to use one flow of Event _or_ ContextChange. Either way I think we should have a test to verify order can be controlled.